lwip-interface: fix issue #2993

0xc0170 opened the pull request:
## Description

Fix for the issue #2993, do not use ``platform.h`` in the C files.

## Status
**READY**

Tested defining EMAC for one platform and  see if DEVICE_ it's properly defined and the code is compiled.

@pan-

pan- commented:
@0xc0170 It fail to compile printing the following error: 

> [ERROR] .\mbed-os\features\FEATURE_LWIP\lwip-interface\emac_lwip.c:17:25: fatal error: mbed_config.h: No such file or directory

0xc0170 commented:
Thanks @pan- , I had old config in the example I was testing, cleaned and reproduced the issue. I rebased it, thus no inclusion required there. Please review again

pan- commented:
@0xc0170 It doesn't work, other parts of lwip (in my case `./mbed-os/hal/hal/emac_api.h`) are pulling `platform.h` unprefixed.

bridadan commented:
/morph test

bridadan commented:
I tried the HeartRate example with this patch on the K64F with the ST BLE shield and it worked. Are you seeing the same thing @pan- ?

bridadan commented:
retest uvisor

0xc0170 commented:
@pan- Need your +1 for this one !

0xc0170 commented:
/morph test

pan- commented:
+1 now.
